### PR TITLE
Preg::split - fix for non-UTF8 subject

### PR DESCRIPTION
**Pull request by kubawerlos**
replaces https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4093

### CODE REVIEW

**Comment by ntzm** [comment]
Shall we use `PREG_NO_ERROR` instead of `0`? More verbose and catches if it's redefined it a later release of PHP (highly unlikely ;) )

**Comment by kubawerlos** [comment]
Ok, now I need some help, I cannot reproduce the errors, all tests are passing for me.

**Comment by kubawerlos** [comment]
Okay, found the reason: https://bugs.php.net/bug.php?id=74183

See the difference: https://3v4l.org/eF5DO#v7032

File: src/Preg.php [hunk 1]
```diff
@@ -36,12 +36,12 @@
     public static function match($pattern, $subject, &$matches = null, $flags = 0, $offset = 0)
     {
         $result = @preg_match(self::addUtf8Modifier($pattern), $subject, $matches, $flags, $offset);
-        if (false !== $result) {
+        if (false !== $result && PREG_NO_ERROR === preg_last_error()) {
```
Comment (ntzm) [inline src/Preg.php]: Perhaps this logic can be split into a private method like `isErroneousResult($result)`?

Comment (kubawerlos) [reply src/Preg.php]: I was thinking about it, but `$result` would have to be compared sometimes to `false` and sometimes to `null` - having both in the private method would be confusing for the reader and having another argument in method call was looking less readable either.

Comment (ntzm) [reply src/Preg.php]: Makes sense

**Comment by SpacePossum** [comment]
Thank you @kubawerlos.

**Comment by SpacePossum** [comment]
Thanks @ostrolucky  as well :)

**Pull request merged**